### PR TITLE
Implement paginated lists in CLI, APIs, and storage

### PR DESCRIPTION
**Pull request by cipherboy**
This implements list paginations using an `after`/`limit` style system, discussed in #140.

In particular, `LIST` API operations are `GET` requests with a `?list=true` parameter to an endpoint. This adds the ability (in the CLI, API, and underlying storage mechanism) to specify two additional attributes, a key, `after`, to show results whose key is strictly greater than (alphabetically) the specified value, and a number, `limit`, of results to show.

This allows better user interface design: pagination can be done on the server (in addition to the UI), which allows the client to consume less resources (both locally and on the remote server), when lots of entries are present. Endpoints like PKI's revoked or valid certificates, Transit's lists of keys, or entries in K/V v2 are all common endpoints upgraded with pagination that benefit from this improvement.

This is an example of a change made possible by dropping support for alternative storage backends. 

Resolves: #140

### CODE REVIEW

**Comment by schultz-is** [comment]
This is great! I can attest to this being a highly desired feature. Notably, KV and PKI backend deployments can get large fast, so the ability to page over all data within them is super handy. The fact that this approach is also backwards compatible (doesn't add any output parameters like cursor paging,) is a nice bonus; that means nobody has to update any existing scripts. Passing the current `List` operations through the `ListPage` operation will also net free test coverage.

File: website/content/docs/internals/telemetry/metrics/all.mdx
Comment (JanMa) [inline website/content/docs/internals/telemetry/metrics/all.mdx:1]: Why are you removing these partials here?

File: builtin/credential/approle/path_role.go [hunk 1]
```diff
@@ -308,6 +308,18 @@ can only be set during role creation and once set, it can't be reset later.`,
 				OperationPrefix: operationPrefixAppRole,
 				OperationSuffix: "roles",
 			},
+			Fields: map[string]*framework.FieldSchema{
+				"after": {
+					Type: framework.TypeString,
+					Description: `Optional entry to list begin listing after,
+not required to exist.`,
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply builtin/credential/approle/path_role.go]: Hmm, thanks. I've updated them, but do note they're inconsistent between plugins (whether or not line wrapping is used). The path-help itself removes them:

https://github.com/openbao/openbao/blob/2c27dde78465c3a633976350b13a8520c27fc657/sdk/framework/path.go#L348

so it ultimately doesn't matter and is mostly a matter of style, if we want a fixed-ish width or want to rely on editor's wrapping :-)

File: website/content/docs/internals/telemetry/metrics/all.mdx
Comment (cipherboy) [reply website/content/docs/internals/telemetry/metrics/all.mdx:1]: Partials were actually removed in #194, but it turns out I didn't remove the references to them :facepalm: This time, when adding the RFC, I apparently _actually_ did the `cd website && make`  I thought I did in that PR, but didn't. :-D 

So this is just to fix the website build, as noted in the PR.

Comment (JanMa) [reply website/content/docs/internals/telemetry/metrics/all.mdx:1]: I see, thanks :-)

**Review by JanMa — APPROVED** [review]

**Pull request merged**